components/dash-core-components/src/components/Checklist.react.js

```diff
@@ -84,36 +77,7 @@ Checklist.propTypes = {
     /**
      * An array of options
      */
-    options: PropTypes.arrayOf(
-        PropTypes.oneOfType([
-            /**
-             * We now accept the single `value` as an option value,
-             * which equals to
-             * { label: `value`, value: `value`, disabled: false }
-             */
-            PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-            PropTypes.exact({
-                /**
-                 * The checkbox's label
-                 */
-                label: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
-                    .isRequired,
-
-                /**
-                 * The value of the checkbox. This value
-                 * corresponds to the items specified in the
-                 * `value` property.
-                 */
-                value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
-                    .isRequired,
-
-                /**
-                 * If true, this checkbox is disabled and can't be clicked on.
-                 */
-                disabled: PropTypes.bool,
-            }),
-        ])
-    ),
+    options: optionsType,
```

Comment:
Although this is a very elegant solution, using variables in this way with PropTypes doesn't work with the Dash toolchain.  The docstrings fail to generate correctly. If you look at the docstring  in `Checklist.py`, you will see that none of the details from `optionsType` are included --  It only shows:

```
- options (optional):
        An array of options.
```
This may be related to #1674, except I didn't see type-errors with the Julia components. I'd love to know if there is a better workaround than just repeating the code. :hocho:  I ran into this in the `dash-bootstrap-components` library as well.

Reply:
That's correct, and yeah there is no better workaround that I know of at this point